test/cmd/debug.sh

```diff
@@ -51,6 +49,26 @@ run_kubectl_debug_pod_tests() {
   # Clean up
   kubectl delete pod target target-copy "${kube_flags[@]:?}"
 
+  # Pre-Condition: Pod "nginx" with labels, annotations, probes and initContainers is created
+  kubectl create -f hack/testdata/pod-with-metadata-and-probes.yaml
+  kube::test::get_object_assert pod "{{range.items}}{{${id_field:?}}}:{{end}}" 'target:'
+  # Command: create a copy of target with a new debug container with --keep-* flags
+  # --keep-* flags intentionally don't work with legacyProfile(Only labels are removed)
+  kubectl debug target -it --copy-to=target-copy --image=busybox --container=debug-container --keep-labels=true --keep-annotations=true --keep-liveness=true --keep-readiness=true --keep-startup=true --keep-init-containers=false --attach=false "${kube_flags[@]:?}"
+  # Post-Conditions
+  kube::test::get_object_assert pod "{{range.items}}{{${id_field:?}}}:{{end}}" 'target:target-copy:'
+  kube::test::get_object_assert pod/target-copy '{{.metadata.labels}}' '<no value>'
+  kube::test::get_object_assert pod/target-copy '{{.metadata.annotations}}' 'map\[test:test\]'
+  kube::test::get_object_assert pod/target-copy '{{range.spec.containers}}{{.name}}:{{end}}' 'target:debug-container:'
+  kube::test::get_object_assert pod/target-copy '{{range.spec.containers}}{{.image}}:{{end}}' "${IMAGE_NGINX:?}:busybox:"
+  kube::test::get_object_assert pod/target-copy '{{range.spec.containers}}{{if (index . "livenessProbe")}}:{{end}}{{end}}' ':'
```

Comment:
What is the expected output in here and why are we checking `:`?

Reply:
It checks whether any containers in target-copy has livenessProbe.
If it dose, go-template return `:`.
I reused an existing template:
https://github.com/kubernetes/kubernetes/blob/8362f51c6620a55953ac564d24fa82d2f2ea2403/test/cmd/debug.sh#L147C1-L147C126

However, if it is not intuitive, I'll fix it.

eg1: return `true` if container has livenessProbe.
```sh
kube::test::get_object_assert pod/target-copy '{{range.spec.containers}}{{if (index . "livenessProbe")}}true{{end}}{{end}}' 'true'
```

eg2: return livenessProbe's exec field value if container has livenessProbe and check them.
```sh
kube::test::get_object_assert pod/target-copy '{{range.spec.containers}}{{.livenessProbe.exec}}:{{end}}' 'map\[command:\[/bin/sh -c cat probe\]\]:<no value>:'
```